src/main/java/org/mskcc/cbio/oncokb/service/UserService.java

```diff
@@ -623,6 +623,11 @@ private void expireUserAccount(UserDTO userDTO){
      * @param userDTO the user
      */
     private Optional<Company> findAssociatedCompany(UserDTO userDTO) {
+        
+        if(userDTO.getCompany() != null){
+            return companyRepository.findById(userDTO.getCompany().getId());
+        }
+
         String domain = StringUtil.getEmailDomain(userDTO.getEmail());
         List<CompanyDomain> companyDomains = companyDomainRepository.findByName(domain);
         if(companyDomains.size() > 0) {
```

Comment:
I just noticed the issue here. When getting the company info by domain, we need to choose the company with Regular license model. If the company is on micro, we cannot simply associate.

Reply:
This will be addressed in the PR for [this issue](https://github.com/oncokb/oncokb/issues/2719)